examples/restriction/const_path_join/src/lib.rs

```diff
@@ -167,6 +236,27 @@ fn is_lit_string(cx: &LateContext<'_>, expr: &Expr<'_>) -> Option<String> {
     }
 }
 
+fn check_component_type<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'tcx>) -> ComponentType {
+    // Check if it's a direct string literal
+    if let Some(s) = is_lit_string(cx, expr) {
+        return ComponentType::Literal(s);
+    }
+
+    // Check if it's a constant-evaluatable string expression
+    if !expr.span.from_expansion()
+        && is_const_evaluatable(cx, expr)
+        && matches!(cx.typeck_results().expr_ty(expr).kind(), ty::Str)
+    {
+        if let Some(snippet) = snippet_opt(cx, expr.span) {
+            // Consider it a constant component if we can get the snippet
+            return ComponentType::Constant(snippet);
+        }
+    }
+
+    // Otherwise, it's neither a literal nor a recognized constant string
+    ComponentType::Other
+}
+
 #[test]
 fn ui() {
     dylint_testing::ui_test_example(env!("CARGO_PKG_NAME"), "ui");
```

Comment:
At face value, your changes are looking pretty good, but I don't think the new test is being run.

I think you need to make a change like the following:
```suggestion
    dylint_testing::ui_test_examples(env!("CARGO_PKG_NAME"));
```

And then in Cargo.toml, add `const_expr` as an example, e.g.:
```toml
[[example]]
name = "ui_const_expr"
path = "ui/const_expr.rs"
```